modules/nw-customize-ingress-error-pages.adoc

```diff
@@ -0,0 +1,104 @@
+// Module filename: nw-customize-ingress-error-pages.adoc
+// Module included in the following assemblies:
+// * networking/ingress-controller-configuration.adoc
+
+[id="nw-customize-ingress-error-pages_{context}"]
+= Customizing HAProxy error code response pages
+
+As a cluster administrator, you can specify a custom error code response page for either 503, 404, or both error pages. The HAProxy router serves a 503 error page when the application pod is not running or a 404 error page when the requested URL does not exist. For example, if you customize the 503 error code response page, then the page is served when the application pod is not running, and the default 404 error code HTTP response page is served by the HAProxy router for an incorrect route or a non-existing route.
+
+Custom error code response pages are specified in a config map then patched to the Ingress Controller. The config map keys have two available file names as follows:
+`error-page-503.http` and `error-page-404.http`.
+
+Custom http error code response pages must follow the link:https://www.haproxy.com/documentation/hapee/latest/configuration/config-sections/http-errors/[HAProxy HTTP error page configuration guidelines]. Here is an example of the default {product-title} HAProxy router link:https://raw.githubusercontent.com/openshift/router/master/images/router/haproxy/conf/error-page-503.http[http 503 error code response page]. You can use the default content as a template for creating your own custom page.
+
+By default, the HAProxy router serves only a 503 error page when the application is not running or when the route is incorrect or non-existent. This default behavior is the same as the behavior on {product-title} 4.8 and earlier. If a config map for the customization of an HTTP error code response is not provided, and you are using a custom HTTP error code response page, the router serves a default 404 or 503 error code response page.
```

Comment:
An "up-to-you": I get it that the 503 is sent when the application isn't running, but I expect a 404 if the route does not exist.  I don't know that any customers will be too concerned if you leave as-is though.

Reply:
Custom Error pages can be a marketing / UX thing; see https://blog.hubspot.com/blog/tabid/6307/bid/33766/10-clever-website-error-messages-from-creative-companies.aspx

Reply:
I feel meh about this detail just because there's so many "If this, then thats" already in here. Too muchies.